app/services/league/controllers/tradeController.js

```diff
@@ -0,0 +1,121 @@
+const fetch = require('node-fetch');
+
+const { Order, Portfolio } = require('../models/portfolioModel');
+const League = require('../models/leagueModel');
+
+const MARKET_ORDERS = ['marketBuy', 'marketSell'];
+
+const getMarketPrice = async (ticker) => {
+    try {
+        const currPrice = await fetch(`${process.env.STOCK_URL}/equity/current/${ticker}`);
+        return currPrice;
+    } catch (err) {
+        console.error(err);
+    }
+};
+
+exports.trade = async (req, res) => {
+    try {
+        let order;
+        let userPortfolio;
+        let currPrice;
+        const { username } = res.locals;
+        const selectedLeague = await League.findOne(
+            { leagueName: req.body.leagueName },
+            (err) => {
+                if (err) throw err;
+            },
+        );
```

Comment:
You can use the findValidLeague middleware to find the league beforehand. Then simply access it through res.locals.league

Reply:
Done